decimalSuite_test.go

```diff
@@ -0,0 +1,138 @@
+		testVal1, testVal2, testAns, parseResult = convertToDec64(testVals[testName], testName)
+		TestPass, TestResult = doTests(testVal1, testVal2, testAns, testVals[testName]["TestFunc"])
+		if TestPass == false {
+			fmt.Printf("%s: \n %s\n %s\n", testName, TestResult, parseResult)
+		}
+	}
+
+}
+
+// func TestNew(t *testing.T) {
+// 	// require := require.New(t)
+// 	// // propper rounding
+// 	// require.Equal(MustParseDecimal64("4444444444444445"), MustParseDecimal64("4444444444444444").Add(MustParseDecimal64("0.5001")))
+// 	// require.Equal(MustParseDecimal64("0.23"), MustParseDecimal64("1.3").Add(MustParseDecimal64("-1.07")))
+// 	// fmt.Println("sjoidgf", MustParseDecimal64("4444444444444444").Add(MustParseDecimal64("1.5001")))
+// 	// // require.Equal(MustParseDecimal64("12345678901234.29"), MustParseDecimal64("12345678901234").Add(MustParseDecimal64("0.2951")))
+//
+// }
+
+//getInput gets the test file and extracts test using regex, then returns a map object and a list of test names
+func getInput(file string) (map[string]map[string]string, []string) {
+	dat, _ := ioutil.ReadFile("dectest/ddAdd.decTest")
+	dataString := string(dat)
+	//  "(\n|\r)") // start with newline
+	//  "(?P<TestName>dd[\w]*)") // TestName must start with dd (double decimal)
+	//  "(?P<TestFunc>[\S]*)") // testfunc made of anything that isn't a whitespace
+	//  "(\s*\'?)") // after can be any number of spaces and quotations if they exist
+	// "(?P<TestVals_1>(\+|-)*[^\r\n\t\f\v\' ]*)") // first test val is anything that isnt a whitespace or a quoteation mark
+	//  "(?P<TestVals_1>(\+|-)*[^\r\n\t\f\v\' ]*)") // first test val is anything that isnt a whitespace or a quoteation mark
+	// ('?\s*'?) match any quotation marks and any spaces
+	// (?P<TestVals_2>(\+|-[^->])?[^\r\n\t\f\v\' ]*) testvals2 same as 1 but no '->'
+	// ('?\s*->\s*'?) matches the indicator to answer
+	// (?P<answer>(\+|-)*[^\r\n\t\f\v\' ]*) matches the answer that's anything that is plus minus but not quotations
+
+	// split into groups: testName, TestFunct, TestVals (x2) and TestAns
+	regex := `(\n|\r)(?P<TestName>dd[\w]*)(\s*)(?P<TestFunc>[\S]*)(\s*\'?)(?P<TestVals_1>(\+|-)*[^\r\n\t\f\v\' ]*)('?\s*'?)(?P<TestVals_2>(\+|-[^->])?[^\r\n\t\f\v\' ]*)('?\s*->\s*'?)(?P<answer>(\+|-)*[^\r\n\t\f\v\' ]*)`
+	r := regexp.MustCompile(regex)
+	ans := r.FindAllStringSubmatch(dataString, -1)
+	testList := []string{}
+	var data = map[string]map[string]string{}
+
+	for i := 0; i < len(ans); i++ {
+		data[ans[i][2]] = make(map[string]string)
+		data[ans[i][2]]["TestFunc"] = ans[i][4]
+		data[ans[i][2]]["TestVal1"] = ans[i][6]
+		data[ans[i][2]]["TestVal2"] = ans[i][9]
+		data[ans[i][2]]["TestAns"] = ans[i][12]
+		testList = append(testList, ans[i][2])
+
+	}
+	return data, testList
+
+}
+
+//convertToDec64 converts the map object strings to decimal64s
+func convertToDec64(testVals map[string]string, testName string) (testVal1, testVal2, testAns Decimal64, parseErr string) {
+	var err1, err2, err3 error
```

Comment:
I don't want to panic because a lot of the times following tests show a similar behaviour that gives extra context to the bug. 
Will follow up the decvals struct